core/frontend/src/render/RenderSystem.ts

```diff
@@ -391,6 +391,9 @@ export abstract class RenderSystem implements IDisposable {
    */
   public abstract createBatch(graphic: RenderGraphic, features: PackedFeatureTable, range: ElementAlignedBox3d, options?: BatchOptions): RenderGraphic;
 
+  /** Return a Promise which when resolved indicates that all pending external textures have finished loading from the backend. */
+  public abstract waitForAllExternalTextures(): Promise<void>;

```

Comment:
Make this non-abstract and make it return `Promise.resolve` so only System has to override it.

Reply:
Done - that makes it much simpler.